core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/keymanager/imports/ImportKeystoresHandler.java

```diff
@@ -0,0 +1,248 @@
+import tech.pegasys.signers.bls.keystore.KeyStoreValidationException;
+import tech.pegasys.signers.bls.keystore.model.KeyStoreData;
+import tech.pegasys.teku.bls.BLSKeyPair;
+import tech.pegasys.teku.bls.BLSSecretKey;
+import tech.pegasys.web3signer.core.multikey.metadata.FileKeyStoreMetadata;
+import tech.pegasys.web3signer.core.multikey.metadata.SignerOrigin;
+import tech.pegasys.web3signer.core.signing.ArtifactSignerProvider;
+import tech.pegasys.web3signer.core.signing.BlsArtifactSigner;
+import tech.pegasys.web3signer.core.signing.KeyType;
+import tech.pegasys.web3signer.slashingprotection.SlashingProtection;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.api.RequestParameters;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+
+public class ImportKeystoresHandler implements Handler<RoutingContext> {
+
+  private static final Logger LOG = LogManager.getLogger();
+  public static final int SUCCESS = 200;
+  public static final int BAD_REQUEST = 400;
+  public static final int SERVER_ERROR = 500;
+  private static final ObjectMapper YAML_OBJECT_MAPPER = new ObjectMapper(new YAMLFactory());
+
+  private final ObjectMapper objectMapper;
+  private final Path keystorePath;
+  private final Optional<SlashingProtection> slashingProtection;
+  private final ArtifactSignerProvider artifactSignerProvider;
+
+  public ImportKeystoresHandler(
+      final ObjectMapper objectMapper,
+      final Path keystorePath,
+      final Optional<SlashingProtection> slashingProtection,
+      final ArtifactSignerProvider artifactSignerProvider) {
+    this.objectMapper = objectMapper;
+    this.keystorePath = keystorePath;
+    this.slashingProtection = slashingProtection;
+    this.artifactSignerProvider = artifactSignerProvider;
+  }
+
+  @Override
+  public void handle(RoutingContext context) {
+    // API spec - https://github.com/ethereum/keymanager-APIs/tree/master/flows#import
+    final RequestParameters params = context.get("parsedParameters");
+    final ImportKeystoresRequestBody parsedBody;
+    try {
+      parsedBody = parseRequestBody(params);
+    } catch (final IllegalArgumentException | JsonProcessingException e) {
+      handleInvalidRequest(context, e);
```

Comment:
Can a test be added for this case? This is different from the invalidRequestBodyReturnsError test case as that fails on the openapi spec verification

Reply:
how is it different? the test passes valid json just not the one expected, so I think it should exercise this part of the code no?

Reply:
The difference is that json that doesn't match the schema is reject with a 400 bad request before the ImportKeystoresHandler even reaches the request.  Don't see how this could ever occur though as to determine whether it matches the schema I think it also needs to be parsed by library.